CredentialProvider.Microsoft/Util/SessionTokenCache.cs

```diff
@@ -48,9 +49,26 @@ public bool ContainsKey(Uri key)
             return Cache.ContainsKey(key);
         }
 
-        public bool TryGetValue(Uri key, out string value)
+        public bool TryGetValue(Uri key, ILogger logger, out string value)
         {
-            return Cache.TryGetValue(key, out value);
+            try
+            {
+                return Cache.TryGetValue(key, out value);
+            }
+            catch (Exception e)
+            {
+                if (File.Exists(cacheFilePath))
+                {
+                    logger.Verbose(string.Format(Resources.CacheException, e.Message));
+
+                    File.Delete(cacheFilePath);
+                    Cache.Remove(key);
```

Comment:
If the cache is corrupted, we should reset the entire Cache object rather than removing just one key.